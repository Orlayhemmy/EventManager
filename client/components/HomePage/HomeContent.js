import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import Welcome from './HomeContent/Welcome';
import { validateSignupInput } from '../../shared/userValidation';
import { userSignupRequest } from '../../actions/signInActions';
import Navbar from '../Navbar.jsx';


/**
 * @description Signin form component
 */
export class HomeContent extends React.Component {
  /**
   * @memberof HomeContent
   * @description it creates an instance of HomeContent
   */
  constructor() {
    super();
    this.state = {
      fullname: '',
      email: '',
      password: '',
      retypePass: '',
      errors: {},
      isLoading: '',
      serverError: '',
      image: '',
      loginEmail: '',
      loginPassword: '',
    }
    this.onChange =this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.isValid = this.isValid.bind(this);
  }
  
  /**
   * @memberof SignUpForm
   * @method onChange
   * @description it sets user input to state
   * @param {object} event
   */
  onChange(e) {
    this.setState({
      [e.target.id]: e.target.value
    })
  }

  /**
   * @memberof SignUpForm
   * @method isValid
   * @description it calls validation action on user data
   * @param {void}
   * @returns true or false
   */
  isValid() {
    const {
      errors,
      isValid
    } = validateSignupInput(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  onImageSelected(event) {
      if (event.target.files && event.target.files[0]) {
          let reader = new FileReader();
  
          reader.onload = (e) => {
            this.setState({
              image: e.target.result
            });
          }
  
          reader.readAsDataURL(event.target.files[0]);
      }
  }
  /**
   * @memberof SignUpForm
   * @method onSubmit
   * @description it calls the user signup action
   * @param {object} event
   * @returns {void}
   */
  onSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
        this.state.title = 'Welcome to Ecenter';
        this.state.message = `Thank you for choosing Ecenter, We hope to make your events
        memorable.<br/> Click on this <a href="#">link</a> to see our event centers and get started`;
      this.props.userSignupRequest(this.state);
    }
  } 

  /**
   * @memberof HomeContent
   * @method render
   * @description it renders the component
   * @returns the HTML of homecontent component
   */
  render() {
    const { isAuth, status } = this.props.auth;
    const {
      fullname,
      email,
      password,
      retypePass,
      errors,
      serverError,
      image,
      loginEmail,
      loginPassword
    } = this.state;
    
    return (
      <div className="container" id="homepage">
        <Navbar path='/' email={this.loginEmail} onChange={this.onChange}/>
        <div className="main-content">
          <div className="row">
            <div className="col-lg-6 intro">
              <h1>Book The Best Center</h1>
              <p>
                Hotel icons, booking and reservation ready and thirteen HTML
                pages, that is roughly said what Colina delivers. Of course,
                there is more. Due to the popularity of mobile users, Colina has
                a mobile-first approach to its web design. But in general, all
                users will experience top-notch and highly adaptable website
                that you are yet to make.
              </p>
              <p>
                Andelaâ€™s business model is simply translating brilliant minds
                into world-class developers and inculcating them into software
                engineering teams around the world, to solve the problem of
                shortage of tech talent and also help achieve quality product
                build. Andela proffers solution to companies or individuals
                which are commonly referred to as partners with a shortage of
                technological team or in need of proficient and experienced
                software personnel by offering them world-class developers to
                solve their challenges or bring their ideas into reality using
                technological tools
              </p>
            </div>
            <div className="col-lg-6 form">
              <h1>Get started</h1>
              <span className="help-block">{this.props.auth.message}</span>
              <form id="signup-form" onSubmit={this.onSubmit}>
              {image === '' ? (
                <div className="imageUpload center">
                  <label for="imageInput">
                    <p className="img-fluid img-circle p-5">
                      Click here to upload your image{' '}
                    </p>
                  </label>
                  <input
                    type="file"
                    id="imageInput"
                    onChange={this.onImageSelected.bind(this)}
                  />
                </div>
                  ) : (
                    <div className="imageUpload center">
                      <label for="imageInput">
                        <img className="img-fluid img-circle" src={image}/>
                      </label>
                      <input
                        type="file"
                        id="imageInput"
                        onChange={this.onImageSelected.bind(this)}
                      />
                    </div>
                  )
                }
                <div className="form-group">
                <span className="help-block">{errors.fullname}</span>
                  <label for="fullname">Fullname</label>
                  <input
                    id="fullname"
                    value={this.state.fullname}
                    placeholder="Fullname"
                    type="text"
                    className="form-control"
                    onChange={this.onChange}
                    required
                  />
                </div>

                <div className="form-group">
                <span className="help-block">{errors.email}</span>
                  <label for="email">Email Address</label>
                  <input
                    id="email"
                    value={this.state.email}
                    placeholder="Email Address"
                    type="email"
                    className="form-control"
                    error={errors.email}
                    onChange={this.onChange}
                    required
                  />
                </div>
                <span className="help-block">{errors.password || errors.retypePass}</span>
                <div className="form-row">
                  <div className="form-group col-md-6">
                    <label for="password">Password</label>
                    <input
                      id="password"
                      value={this.state.password}
                      placeholder="Password"
                      type="password"
                      className="form-control"
                      error={errors.password}
                      onChange={this.onChange}
                      required
                    />
                  </div>
                  <div className="form-group col-md-6">
                    <label for="retypePass">Retype-Password</label>
                    <input
                      id="retypePass"
                      value={this.state.retypePass}
                      placeholder="Re-type Password"
                      type="password"
                      className="form-control"
                      onChange={this.onChange}
                      required
                    />
                  </div>
                </div>
                <div className="center">
                  <input
                    id="signup"
                    type="submit"
                    value="Join Us"
                    className="btn btn-primary mb-2"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
        <div className="row subcontent">
          <div className="col-lg-4">
            <div className="row center">
              <span class="icon">
                <i className="fa fa-dollar" />
              </span>
            </div>
            Hotel icons, booking and reservation ready and thirteen HTML pages,
            that is roughly said what Colina delivers. Of course, there is more.
            Due to the popularity of mobile users, Colina has a mobile-first
            approach to its web design. But in general, all users will
            experience top-notch and highly adaptable website that you are yet
            to make.
          </div>
          <div className="col-lg-4">
            <div className="row center">
              <span class="icon">
                <i className="fa fa-lock" />
              </span>
            </div>
            Hotel icons, booking and reservation ready and thirteen HTML pages,
            that is roughly said what Colina delivers. Of course, there is more.
            Due to the popularity of mobile users, Colina has a mobile-first
            approach to its web design. But in general, all users will
            experience top-notch and highly adaptable website that you are yet
            to make.
          </div>
          <div className="col-lg-4">
            <div className="row center">
              <span class="icon">
                <i className="fa fa-thumbs-up" />
              </span>
            </div>
            Hotel icons, booking and reservation ready and thirteen HTML pages,
            that is roughly said what Colina delivers. Of course, there is more.
            Due to the popularity of mobile users, Colina has a mobile-first
            approach to its web design. But in general, all users will
            experience top-notch and highly adaptable website that you are yet
            to make.
          </div>
        </div>
      </div>
    );
  }
}

const propTypes = {
  auth: PropTypes.object.isRequired,
  userSignupRequest: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

HomeContent.propTypes = propTypes;

export default connect(mapStateToProps, {userSignupRequest})(HomeContent);
