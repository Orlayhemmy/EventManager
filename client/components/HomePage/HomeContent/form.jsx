import React from 'react';

export default class Form extends React.Component {
  constructor() {
    super();
    this.state = {
      showMe: false
    }
    this.signUp = this.signUp.bind(this);
    this.signIn = this.signIn.bind(this);
  }

  signIn() {
    this.setState({
      showMe : true
    });
  }

  signUp() {
    this.setState({
      showMe: false
    })
  };

  render() {
    if(this.state.showMe) {
      return (
      <div className="col-lg-4">
        <div className="form-outer text-center">
          <div className="form-inner">
            <div className="logo text-uppercase"><strong className="text-primary">Sign In</strong></div>
            <h2>Welcome Back!</h2>
            <form id="login-form">
              <div className="form-group">
                <input id="signin-email" type="email" placeholder="Email" required/>
              </div>
              <div className="form-group">
                <input id="signin-password" type="password" placeholder="Password" required/>
              </div>
              <input id="login" type="submit" value="login" className="btn btn-primary"/>
            </form>
            <a href="#" className="goto">Forgot Password? Click Here</a><br/>
            <a onClick={this.signUp} href="#" className="goto">are you new here? Create An Account</a>
          </div>
        </div>
      </div>
      );
    } else {
      return (
        <div className="col-lg-4">
          <div className="form-outer text-center"> 
            <div className="form-inner">
              <span className="logo text-uppercase"><strong className="text-primary">sign up</strong></span>
              <h2>Please fill in your details to get started</h2>
              <form id="signup-form">
                <div className="form-group">
                  <input id="username" type="text" placeholder="Fullname" required/>
                </div>
                <div className="form-group">
                  <input id="signup-email" type="email" placeholder="Email" required/>
                </div>
                <div className="form-group">
                  <input id="signup-password" type="password" placeholder="Password" required/>
                </div>
                <div className="form-group">
                  <input id="signup-retypepass" type="password" placeholder="Retype Password" required/>
                </div>
                <input id="signup" type="submit" value="Create Account" className="btn btn-primary"/>
              </form>
              <a onClick={this.signIn} href="#" className="goto">already signed up? login to your account</a>
            </div>
          </div>
        </div>
      );
    }
  }
}