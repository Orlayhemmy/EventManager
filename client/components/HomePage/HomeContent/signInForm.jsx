import React from 'react';
import PropTypes from 'prop-types';

import TextField from '../../../common/textField';
import validation from '../../../shared/userSignInValidation';

class SignInForm extends React.Component {
  constructor() {
    super();

    this.state = {
      email: '',
      password: '',
      errors: {},
      serverError: '',
      isLoading: false
    }

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.isValid = this.isValid.bind(this);
  }

  isValid() {
    const { errors, isValid } = validation(this.state);
    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }

  onChange(e) {
    this.setState({
      [e.target.id]: e.target.value
    });
  }

  onSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {}, isLoading: true });
      this.props.userSignInRequest(this.state).then(() => {
        this.props.addFlashMessage({
            type: 'Success',
            text: 'Successfully Signed In.'
        });
        this.context.router.history.push('/dashboard')
      })
      .catch((error) => {
        this.setState({ serverError: error.response.data.message });
        this.setState({ errors: error.response.data, isLoading: false})
      });
    }

  }

  render() {
    const { email, password, errors, serverError } = this.state;
    return (
    <div className="col-lg-4">
      <div className="form-outer text-center">
        <div className="form-inner">
          <div className="logo text-uppercase"><strong className="text-primary">Sign In</strong></div>
          <h2>Welcome Back!</h2>
          <span className="help-block">{serverError}</span>
          <form id="login-form" onSubmit={this.onSubmit}>
          <TextField
                id='email'
                value={this.state.email}
                placeholder='email address'
                type='email'
                error={errors.email}
                onChange={this.onChange} />
                
                <TextField
                id='password'
                value={this.state.password}
                placeholder='password'
                type='password'
                error={errors.password} 
                onChange={this.onChange} />

            <input id="login" type="submit" value="login" className="btn btn-primary" disabled={this.state.isLoading}/>
          </form>
          <a href="#" className="goto">Forgot Password? Click Here</a><br/>
          <a href="#" className="goto">are you new here? Create An Account</a>
        </div>
      </div>
    </div>
    );
  }
}

SignInForm.propTypes = {
  userSignInRequest: PropTypes.func.isRequired,
}

SignInForm.contextTypes = {
  router: PropTypes.object.isRequired
};

export default SignInForm;
