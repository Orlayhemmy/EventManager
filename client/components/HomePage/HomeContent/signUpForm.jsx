import React from 'react';
import PropTypes from 'prop-types';

import { validateSignupInput } from '../../../shared/userValidation';
import TextField from '../../../common/textField3';




class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      fullname: '',
      email: '',
      password: '',
      retypePass: '',
      errors: {},
      isLoading: '',
      serverError: '',
    }
    this.onChange =this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.isValid = this.isValid.bind(this);
  }
  
  onChange(e) {
    this.setState({
      [e.target.id]: e.target.value
    })
  }

  isValid() {
    const { errors, isValid } = validateSignupInput(this.state);
    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }


  onSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
      
      this.setState({ errors: {}, isLoading: true });
      this.props.userSignupRequest(this.state).then(() => {
        this.props.addFlashMessage({
            type: 'Success',
            text: 'Successfully Created Account.'
        });
        let title = 'Welcome to Ecenter';
        let message = `Thank you for choosing Ecenter, We hope to make your events
        memorable.<br/> Click on this <a href="#">link</a> to see our event centers and get started`
        this.props.sendMail(this.state.email, message, title);
        this.context.router.history.push('/dashboard')
      })
      .catch((error) => {
        this.setState({ serverError: error.response.data.message });
        this.setState({ errors: error.response.data, isLoading: false})
      });
    }
  
  }
 
 

  render() {
    const {
      fullname,
      email,
      password,
      retypePass,
      errors,
      serverError
    } = this.state;
    return (
      <div>
        <div className="logo text-uppercase"><strong className="text-primary">Sign Up</strong></div>          
        <h2>Please fill in your details to get started</h2>
        <form id="signup-form" onSubmit={this.onSubmit}>
          <TextField
            id='fullname'
            value={this.state.fullname}
            placeholder='Fullname'
            type='text'
            error={errors.fullname} 
            onChange={this.onChange} />
            
            <TextField
            id='email'
            value={this.state.email}
            placeholder='Email Address'
            type='email'
            error={errors.email, serverError}
            onChange={this.onChange} />
            
            <TextField
            id='password'
            value={this.state.password}
            placeholder='Password'
            type='password'
            error={errors.password} 
            onChange={this.onChange} />
            
            <TextField
            id='retypePass'
            value={this.state.retypePass}
            placeholder='Re-type Password'
            type='password'
            error={errors.retypePass} 
            onChange={this.onChange} />

          <input id="signup" type="submit" value="Create Account" className="btn btn-primary"/>
        </form>     
      </div>
    )
  }
}

SignUpForm.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  addFlashMessage: PropTypes.func.isRequired,
  sendMail: PropTypes.func.isRequired,
}

SignUpForm.contextTypes = {
  router: PropTypes.object.isRequired
};

export default SignUpForm;
