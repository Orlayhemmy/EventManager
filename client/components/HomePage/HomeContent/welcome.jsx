import React from 'react';
import { Link } from 'react-router-dom';

export default class Welcome extends React.Component {
  render() {
    return (
      <div className="col-lg-8">
        <div className="jumbotron text-center">
          <h1>event center</h1>
          <Link to="/imageupload"><h2>...celebrate in style...</h2></Link>
        </div>
      </div>
    );
  }
}