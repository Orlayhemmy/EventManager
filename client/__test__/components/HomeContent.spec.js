import React from 'react';
import renderer from 'react-test-renderer';
import expect from 'expect';
import HomeContent from '../../components/HomePage/Template/Content/HomeContent';

describe('Home Component', () => {
  const props = {
    centerState: {
      centers: ['balmoral', 'andela']
    },
    isAuth: true
  };
  it('should render without throwing error', () => {
    const wrapper = renderer.create(<HomeContent />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('show carousel when isAuth value is true', () => {
    const wrapper = renderer.create(<HomeContent isAuth={props.isAuth} centerState={props.centerState}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
