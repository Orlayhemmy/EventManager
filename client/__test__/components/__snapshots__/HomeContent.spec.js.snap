// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should carousel when value is true 1`] = `
<div
  id="homepage"
>
  <div
    className="row main-content-bg"
  >
    <div
      className="main-content container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-6 intro"
        >
          <h1>
            Book The Best Center
          </h1>
          <p>
            Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
          </p>
          <p>
            Andela’s business model is simply translating brilliant minds into world-class developers and inculcating them into software engineering teams around the world, to solve the problem of shortage of tech talent and also help achieve quality product build. Andela proffers solution to companies or individuals which are commonly referred to as partners with a shortage of technological team or in need of proficient and experienced software personnel by offering them world-class developers to solve their challenges or bring their ideas into reality using technological tools
          </p>
        </div>
        <div
          className="col-lg-6 form"
        >
          <h1>
            Our Centers
          </h1>
          <div
            className="carousel slide"
            data-ride="carousel"
            id="carouselExampleIndicators"
          >
            <ol
              className="carousel-indicators"
            >
              <li
                className="active"
                data-slide-to={0}
                data-target="#carouselExampleIndicators"
              />
              <li
                className=""
                data-slide-to={1}
                data-target="#carouselExampleIndicators"
              />
            </ol>
            <div
              className="carousel-inner"
            >
              <div
                className="carousel-item active"
              >
                <img
                  alt={undefined}
                  className="d-block w-100 h-350"
                  src={undefined}
                />
                <div
                  className="carousel-caption d-none d-md-block"
                >
                  <h5 />
                  <p />
                </div>
              </div>
              <div
                className="carousel-item"
              >
                <img
                  alt={undefined}
                  className="d-block w-100 h-350"
                  src={undefined}
                />
                <div
                  className="carousel-caption d-none d-md-block"
                >
                  <h5 />
                  <p />
                </div>
              </div>
            </div>
            <a
              className="carousel-control-prev"
              data-slide="prev"
              href="#carouselExampleIndicators"
              role="button"
            >
              <span
                aria-hidden="true"
                className="carousel-control-prev-icon"
              />
              <span
                className="sr-only"
              >
                Previous
              </span>
            </a>
            <a
              className="carousel-control-next"
              data-slide="next"
              href="#carouselExampleIndicators"
              role="button"
            >
              <span
                aria-hidden="true"
                className="carousel-control-next-icon"
              />
              <span
                className="sr-only"
              >
                Next
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row sub-color"
  >
    <div
      className="row subcontent"
    >
      <div
        className="row pb-4 w-85"
      >
        <div
          className="col-lg-4 pb-3 justify-text"
        >
          <div
            className="row center"
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-dollar-sign"
              />
            </span>
          </div>
          <p>
            Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
          </p>
        </div>
        <div
          className="col-lg-4 pb-3 justify-text"
        >
          <div
            className="row center"
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-lock"
              />
            </span>
          </div>
          Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
        </div>
        <div
          className="col-lg-4 pb-3 justify-text"
        >
          <div
            className="row center"
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-thumbs-up"
              />
            </span>
          </div>
          Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Component should render without throwing error 1`] = `
<div
  id="homepage"
>
  <div
    className="row main-content-bg"
  >
    <div
      className="main-content container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-6 intro"
        >
          <h1>
            Book The Best Center
          </h1>
          <p>
            Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
          </p>
          <p>
            Andela’s business model is simply translating brilliant minds into world-class developers and inculcating them into software engineering teams around the world, to solve the problem of shortage of tech talent and also help achieve quality product build. Andela proffers solution to companies or individuals which are commonly referred to as partners with a shortage of technological team or in need of proficient and experienced software personnel by offering them world-class developers to solve their challenges or bring their ideas into reality using technological tools
          </p>
        </div>
        <div
          className="col-lg-6 form"
        >
          <h1>
            Get started
          </h1>
          <form
            id="signup-form"
            onSubmit={undefined}
          >
            <span
              className="help-block"
            />
            <div
              className="form-group"
            >
              <label
                htmlFor="fullname"
              >
                Fullname
              </label>
              <input
                className="form-control"
                id="fullname"
                onChange={undefined}
                placeholder="Fullname"
                required={true}
                type="text"
                value={undefined}
              />
            </div>
            <span
              className="help-block"
            />
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Email Address
              </label>
              <input
                className="form-control"
                id="email"
                onChange={undefined}
                placeholder="Email Address"
                required={true}
                type="email"
                value={undefined}
              />
            </div>
            <span
              className="help-block"
            />
            <div
              className="form-row"
            >
              <div
                className="form-group col-md-6"
              >
                <label
                  htmlFor="password"
                >
                  Password
                </label>
                <input
                  className="form-control"
                  id="password"
                  onChange={undefined}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value={undefined}
                />
              </div>
              <div
                className="form-group col-md-6"
              >
                <label
                  htmlFor="retypePass"
                >
                  Retype-Password
                </label>
                <input
                  className="form-control"
                  id="retypePass"
                  onChange={undefined}
                  placeholder="Re-type Password"
                  required={true}
                  type="password"
                  value={undefined}
                />
              </div>
            </div>
            <div
              className="center"
            >
              <input
                className="btn btn-primary mb-2"
                id="signup"
                type="submit"
                value="Join Us"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row sub-color"
  >
    <div
      className="row subcontent"
    >
      <div
        className="row pb-4 w-85"
      >
        <div
          className="col-lg-4 pb-3 justify-text"
        >
          <div
            className="row center"
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-dollar-sign"
              />
            </span>
          </div>
          <p>
            Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
          </p>
        </div>
        <div
          className="col-lg-4 pb-3 justify-text"
        >
          <div
            className="row center"
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-lock"
              />
            </span>
          </div>
          Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
        </div>
        <div
          className="col-lg-4 pb-3 justify-text"
        >
          <div
            className="row center"
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-thumbs-up"
              />
            </span>
          </div>
          Hotel icons, booking and reservation ready and thirteen HTML pages, that is roughly said what Colina delivers. Of course, there is more. Due to the popularity of mobile users, Colina has a mobile-first approach to its web design. But in general, all users will experience top-notch and highly adaptable website that you are yet to make.
        </div>
      </div>
    </div>
  </div>
</div>
`;
